qemu/exec.c

```diff
@@ -1100,6 +1098,9 @@ RAMBlock *qemu_ram_alloc_from_ptr(struct uc_struct *uc, ram_addr_t size, void *h
     new_block->host = host;
     if (host) {
         new_block->flags |= RAM_PREALLOC;
+    } else {
+        size = HOST_PAGE_ALIGN(uc, size);
```

Comment:
The fix looks promising but assigning values to `size` and `max_size` seems doing nothing?

Reply:
Yes, you're right. The max_length assignment should have moved down as well.